cmd/asset-syncer/server/utils.go

```diff
@@ -300,17 +300,102 @@ func (r *HelmRepo) Charts(ctx context.Context, fetchLatestOnly bool, chartResult
 // FetchFiles retrieves the important files of a chart and version from the repo
 func (r *HelmRepo) FetchFiles(cv models.ChartVersion, userAgent string, passCredentials bool) (map[string]string, error) {
 	authorizationHeader := ""
-	chartTarballURL := chartTarballURL(r.AppRepositoryInternal, cv)
+	chartTarballURL, err := url.Parse(chartTarballURL(r.AppRepositoryInternal, cv))
+	if err != nil {
+		return nil, err
+	}
 
-	if passCredentials || len(r.AuthorizationHeader) > 0 && isURLDomainEqual(chartTarballURL, r.URL) {
+	if passCredentials || len(r.AuthorizationHeader) > 0 && isURLDomainEqual(chartTarballURL.String(), r.URL) {
 		authorizationHeader = r.AuthorizationHeader
 	}
 
-	return tarutil.FetchChartDetailFromTarballUrl(
-		chartTarballURL,
-		userAgent,
-		authorizationHeader,
-		r.netClient)
+	// If URL points to an OCI chart, we transform its URL to its tgz blob URL
+	if chartTarballURL.Scheme == "oci" {
+		return FetchChartDetailFromOciUrl(chartTarballURL, userAgent, authorizationHeader, r.netClient)
+	} else {
+		return FetchChartDetailFromTarballUrl(chartTarballURL, userAgent, authorizationHeader, r.netClient)
+	}
+}
+
+// Fetches helm chart details from a gzipped tarball
+//
+// name is expected in format "foo/bar" or "foo%2Fbar" if url-escaped
+func FetchChartDetailFromTarballUrl(chartTarballURL *url.URL, userAgent string, authz string, netClient *http.Client) (map[string]string, error) {
+	reqHeaders := make(map[string]string)
+	if len(userAgent) > 0 {
+		reqHeaders["User-Agent"] = userAgent
+	}
+	if len(authz) > 0 {
+		reqHeaders["Authorization"] = authz
+	}
+
+	// use our "standard" http-client library
+	reader, _, err := httpclient.GetStream(chartTarballURL.String(), netClient, reqHeaders)
+	if reader != nil {
+		defer reader.Close()
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	return tarutil.FetchChartDetailFromTarball(reader)
+}
+
+func FetchChartDetailFromOciUrl(chartTarballURL *url.URL, userAgent string, authorizationHeader string, netClient *http.Client) (map[string]string, error) {
```

Comment:
`authorizationHeader` argument is called `authz` in the `FetchChartDetailFromTarballUrl` function, use one or the other